patches/rust-nix/0001-managarm-initial-port.patch

```diff
@@ -604,28 +764,62 @@ index 36a3601..c57bd1b 100644
 +libc_bitflags! {
      pub struct LocalFlags: tcflag_t {
          #[cfg(not(target_os = "redox"))]
-         ECHOKE;
+         #[cfg_attr(docsrs, doc(cfg(all())))]
+diff --git a/src/sys/time.rs b/src/sys/time.rs
+index c1ab67b..f43cf63 100644
+--- a/src/sys/time.rs
++++ b/src/sys/time.rs
+@@ -16,6 +16,7 @@ const TIMESPEC_ZERO: libc::timespec = unsafe {
+             target_os = "freebsd",
+             target_os = "illumos",
+             target_os = "linux",
++            target_os = "managarm",
+             target_os = "netbsd"
+         ),
+         feature = "time",
+@@ -82,7 +83,7 @@ pub(crate) mod timer {
+         Interval(TimeSpec),
+     }
+ 
+-    #[cfg(any(target_os = "android", target_os = "linux"))]
++    #[cfg(any(target_os = "android", target_os = "linux", target_os = "managarm"))]
+     bitflags! {
+         /// Flags that are used for arming the timer.
+         pub struct TimerSetTimeFlags: libc::c_int {
+diff --git a/src/sys/timerfd.rs b/src/sys/timerfd.rs
+index 18acbae..0005b24 100644
+--- a/src/sys/timerfd.rs
++++ b/src/sys/timerfd.rs
+@@ -57,7 +57,7 @@ impl FromRawFd for TimerFd {
+ libc_enum! {
+     /// The type of the clock used to mark the progress of the timer. For more
+     /// details on each kind of clock, please refer to [timerfd_create(2)](https://man7.org/linux/man-pages/man2/timerfd_create.2.html).
+-    #[repr(i32)]
++    #[repr(i64)]
```

Comment:
This is not really a robust change, but something we'll have to deal with when we upstream :^)